include/xtypes/UnionType.hpp

```diff
@@ -38,8 +38,20 @@ class ReadableDynamicDataRef;
 class WritableDynamicDataRef;
 
 static const std::string UNION_DISCRIMINATOR("discriminator");
-static const int64_t DEFAULT_UNION_LABEL = std::numeric_limits<int64_t>::max();
-static const int64_t INVALID_UNION_LABEL = DEFAULT_UNION_LABEL - 1;
+
+constexpr int64_t default_union_label(
+        size_t type_size)
+{
+    return std::numeric_limits<int64_t>::max() >> ((sizeof(int64_t) - type_size) * 8);
+}
+
+template<typename T>
+constexpr int64_t shrink_label(
+        T value,
+        size_t type_size)
+{
+    return static_cast<int64_t>(value) & (std::numeric_limits<uint64_t>::max()  >> ((sizeof(int64_t) - type_size) * 8));
+}
```

Comment:
Taking into account that the maximum size of any label type is going to be an 8-octet type (uint64_t or int64_t), is this function really necessary?
The allowed types are: bool, [u]int[8,16,32,64]_t, char, and char, this is, all primitive_types except floating point.